src/util/models.py

```diff
@@ -421,7 +421,10 @@ def merge_model(cr, source, target, drop_table=True, fields_mapping=None, ignore
     cr.execute("SELECT model, id FROM ir_model WHERE model in %s", ((source, target),))
     model_ids = dict(cr.fetchall())
     mapping = {model_ids[source]: model_ids[target]}
+
     ignores = ["ir_model", "ir_model_fields", "ir_model_constraint", "ir_model_relation"]
+    ignores.extend(ir.table for ir in indirect_references(cr, bound_only=True) if ir.res_model_id)
```

Comment:
I'd prefer to not alter `ignores`. 

See, below in line 494
```py
if ir.res_model and not ir.res_id and ir.table not in ignores:
```
`bound_only=True` is the same as indirect references with `res_id` set, thus we are still ignoring them anyway.

IMO it's much better for debugging to keep the input `ignores` untouched and we know from experience that overriding variables could have unintended purposes.